holmes/core/conversations_worker/realtime_manager.py

```
@@ -0,0 +1,573 @@
+                # Detect channel closure (e.g. token expiry → server sends
+                # phx_close). The library's auto-reconnect is unreliable
+                # after a channel close, so we do a full teardown/reconnect.
+                if await self._channel_needs_reconnect():
+                    logging.warning(
+                        "Channel no longer joined (state=%s), reconnecting",
+                        self._channel.state if self._channel else "None",
+                    )
+                    self._connected = False
+                    try:
+                        self.on_new_pending()
+                    except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)